### ISSUE TITLE
pFUnit sometimes generates too long lines

### ISSUE THREAD

**Issue by olebole** [Goddard-Fortran-Ecosystem/pFUnit#221]
In our project, I sometimes see the following error with gfortran:

```
Processing file test_config.pf
 ... Done.  Results in test_config.F90
gfortran -g -c […] test_config.F90
[…]/tests/test_config.pf:211:132:

Error: Line truncated at (1) [-Werror=line-truncation]
[…]
```
While one could increase the allowed line length with `-ffree-line-length`, it would also be good to stick with the standard here.

**Comment by tclune** [comment]
I agree that the warnings are annoying and would prefer to stay within the standard.      I've done several things to minimize this.   E.g, the expansion of ```@assertEqual()``` splits the Fortran arguments across lines with one argument per line.    

Generally the longest lines in the generated code are the ```#line``` semaphores that enable more useful line numbers in the original .pf source when the compiler generates errors.

I poked around quickly in some of my projects to see if I had any long lines, but did not find any.   Would you be willing to post a small example (pre and post the parsing step).  I'll be happy to use that to see if I can find any place to further trip the lines.

**Comment by bd4** [comment]
I am running into this with MAX_ASSERT_RANK=6 and gcc-9.1.0

**Comment by tclune** [comment]
@bd4 could you try editing `./cmake/GNU.cmake` and change the `512` on line 9 to be `0`?    If that fixes it for you,
I'll push a fix.   

I like sticking to the standard, but code generators are harder to fix than line-lengths.  Besides, the next Fortran release is going to all extremely long lines and this issue will vanish.

**Comment by bd4** [comment]
Unfortunately that did not fix it, because the externals also set it to 512 and that seems to override the one set in top level cmake/GNU.cmake. Setting `-DCMAKE_Fortran_FLAGS="-ffree-line-length-none"` on the cmake command line does work, so I have a workaround.

**Comment by tclune** [comment]
OK - I can propagate across the whole set of repos this weekend.  (I hope.)

**Comment by tclune** [comment]
I released v4.2.2 today which includes unlimited line length for gfortran in all the dependencies.   Please confirm whether this release fixes your issue.

**Comment by bd4** [comment]
Still failing for min in v4.2.2 with MAX_ASSERT_RANK=6:
```
/home/ci/src/pFUnit/build-gcc9/extern/fArgParse/extern/gFTL-shared/extern/gFTL/include/v1/templates/vector_impl.inc:449:132:

  449 |           __TYPE_MOVE(__GET(this%elements(i+1)),__GET(this%elements(i)))
      |                                                                        
``` 

Workaround `-DCMAKE_Fortran_FLAGS="-ffree-line-length-none"` does still work.

**Comment by mwaxmonsky** [comment]
Apologies for the bother all, but we ran into this kind of issue ourselves using v4.15.0 and noticed that we set `CMAKE_BUILD_TYPE` to `RelWithDebInfo` in our `CMakeLists.txt` instead of `Debug` or `Release` and were propagating that down to our dependencies.

The `GNU.cmake` sets `CMAKE_Fortran_FLAGS_DEBUG` and `CMAKE_Fortran_FLAGS_RELEASE` explicitly but not `CMAKE_Fortran_FLAGS_RELWITHDEBINFO` or `CMAKE_Fortran_FLAGS_MINSIZEREL`.  A little experimenting shows setting the cmake build type specific variable solves the issue but what the proper flags for the config are or if pfunit should support `RelWithDebInfo` or `MinSizeRel` at all are probably for a different issue.

Apologies again if this wasn't the root issue for the above error but I figured I'd leave this here for other users to experiment with if they come across this with a similar problem.
